agent/lib/kontena/workers/event_worker.rb

```diff
@@ -34,9 +34,11 @@ def stream_events
       filters = JSON.dump({type: ['container']})
       defer {
         begin
-          Docker::Event.stream({filters: filters}) do |event|
-            raise "stop event stream" unless processing?
-            @event_queue << event
+          while processing?
+            Docker::Event.stream({filters: filters}) do |event|
+              raise "stop event stream" unless processing?
+              @event_queue << event
+            end
```

Comment:
Maybe log something when the stream closes. Just so that we see how often this happens.